### PR TITLE
MOTECH-931: Documentation for configuring MOTECH to be HIPAA compliant

### PR DESCRIPTION
**Pull request by pgesek**
This doc focuses primarily on the security rule - technical safeguards in particular, since they apply to the MOTECH part of an implementation. The privacy rule is mentioned. I do not dive into adminstrative
stuff and other requirements.

### CODE REVIEW

**Comment by pgesek** [comment]
test this please

File: docs/source/get_started/configure_hipaa.rst [hunk 1]
```diff
@@ -1,5 +1,242 @@
...
+    :depth: 4
+
+What is HIPAA?
+==============
+
+From http://www.hhs.gov/ocr/privacy/hipaa/understanding/srsummary.html:
+
+The Health Insurance Portability and Accountability Act of 1996 (HIPAA) required the Secretary of the U.S.
+Department of Health and Human Services (HHS) to develop regulations protecting the privacy and security of certain
+health information.1 To fulfill this requirement, HHS published what are commonly known as the HIPAA Privacy Rule and
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: Remove "1" annotation

File: docs/source/get_started/configure_hipaa.rst [hunk 2]
```diff
@@ -1,5 +1,242 @@
...
+A unofficial version that presents all regulatory standards in one document (as of March 2013) is available on the website of the
+U.S. Department of Health & Human services:
+
+http://www.hhs.gov/ocr/privacy/hipaa/administrative/combined/hipaa-simplification-201303.pdf
+
+Scope of this document
+======================
+
+Compliance with HIPAA requires implementing physical and technical safeguards, as well as technical and administrative
+policies when it comes to dealing with sensitive patient. It is impossible for us to help you achieve full compliance,
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: Did you mean "sensitive patient data"?

File: docs/source/get_started/configure_hipaa.rst [hunk 3]
```diff
@@ -1,5 +1,242 @@
...
+allows the covered entity to adopt an alternative measure that achieves the purpose of the standard, if the alternative
+measure is reasonable and appropriate.
+
+Following are the HIPAA standards that must be adhered to in order to be HIPAA compliant.
+
+########################
+Standard: Access control
+########################
+
+To be in accordance with standard, only authorized persons or software programs should have access to e-PHI. MOTECH includes
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: people

File: docs/source/get_started/configure_hipaa.rst [hunk 4]
```diff
@@ -1,5 +1,242 @@
...
+user accounts and associate them with different access roles. What you have to do however, is to make sure that all the
+users accessing the system have their own unique accounts with correct access permissions - remember that only authorized persons
+can have access to e-PHI.
+
+Since MOTECH also provides a configurable security rule system, you must make sure that no parts of the system allowing
+any kind of access to ePHI can be accessed without authentication and authorization.
+
+If you are implementing your own modules, you must make sure to secure all HTTP endpoints that require specific permissions
+using Spring Security measures, preferably annotations but other ways of doing it are also acceptable. If you add a new endpoint
+without adding any security safeguards, then by default it still we restricted to only authenticated users (all authenticated users,
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: will be restricted

File: docs/source/get_started/configure_hipaa.rst [hunk 5]
```diff
@@ -1,5 +1,242 @@
...
+
+Refer to the :doc:`Security Model </architecture/security_model>` documentation for more information on this setting.
+
+Encryption and decryption (Addressable)
+#######################################
+
+Implement a mechanism to encrypt and decrypt electronic protected health information.
+
+MOTECH does not provide any out of the box solution for this. It is up to you to either configure encryption and you have
+a few options for achieving this. The first one is to this manually in your code - you would have manually encrypt
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: would have to

File: docs/source/get_started/configure_hipaa.rst [hunk 6]
```diff
@@ -1,5 +1,242 @@
...
+`Article <http://www.porticor.com/2012/05/mysql-cloud-encryption/>`_.
+
+########################
+Standard: Audit controls
+########################
+
+Implement hardware, software, and/or procedural mechanisms that record and examine activity in information systems that
+contain or use electronic protected health information.
+
+If you are using MOTECH Data Services for persistence, then you are given a history recording feature for you data out of box.
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: your

File: docs/source/get_started/configure_hipaa.rst [hunk 7]
```diff
@@ -1,5 +1,242 @@
...
+It is not enabled for new entities by default, so you will have to turn on it either through UI or the @Entity annotation.
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: turn it on

File: docs/source/get_started/configure_hipaa.rst [hunk 8]
```diff
@@ -1,5 +1,242 @@
...
+The history feature will track all changes to an entity and record them in a separate table of the table. This history
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: "table of the table" :)

File: docs/source/get_started/configure_hipaa.rst [hunk 9]
```diff
@@ -1,5 +1,242 @@
...
+The last login date for each user is recorded and stored in the database.
+
+Moreover MOTECH implements a comprehensive logging system based on SLF4J that allows you to track the functioning of
+the system. The suggest logging level for operations is INFO, but you can tweak the logging levels of different components
+as you see fit, even during runtime.
+
+A status message API for posting status messages that get persisted in the database is also exposed by the admin module
+and can be leveraged.
+
+When you develop you own application, you make sure you have enough logging statements to trace activity in the
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: your

File: docs/source/get_started/configure_hipaa.rst [hunk 10]
```diff
@@ -1,5 +1,242 @@
...
+that is being transmitted over an electronic communications network.
+
+Integrity controls (Addressable)
+################################
+
+Implement security measures to ensure that electronically transmitted electronic protected health information
+is not improperly modified without detection until disposed of.
+
+This means using encryption for all e-PHI data being sent over the wire. This includes, but is not limited to HTTP
+communication, database communication (if its over a network), emails and so forth. Since MOTECH runs on Tomcat,
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: it's

File: docs/source/get_started/configure_hipaa.rst [hunk 11]
```diff
@@ -1,5 +1,242 @@
...
+for sending emails, outside systems such as OpenMRS or Commcare and so on. When using these services that are not the same
+host as the MOTECH server, make sure that use `HTTPS` instead of plain `HTTP` or in other cases, such as the email server,
+make sure that proper cryptographic configuration is being used - this is not configured out of the box.
+
+Encryption (Addressable)
+########################
+
+Implement a mechanism to encrypt electronic protected health information whenever deemed appropriate.
+
+As mentioned, MOTECH does not provide out of the box encryption utils. You can implement you own encryption methods using
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: your

File: docs/source/get_started/configure_hipaa.rst [hunk 12]
```diff
@@ -1,5 +1,242 @@
...
+================
+
+The HIPAA Privacy Rule establishes national standards to protect individuals’ medical records and other personal
+health information and applies to health plans, health care clearinghouses, and those health care providers that conduct
+certain health care transactions electronically.  The Rule requires appropriate safeguards to protect the privacy of
+personal health information, and sets limits and conditions on the uses and disclosures that may be made of such
+information without patient authorization. The Rule also gives patients rights over their health information,
+including rights to examine and obtain a copy of their health records, and to request corrections.
+
+It is up to you as an implementer and an entity covered by HIPAA to adhere to the privacy rule. You have take care
```
Comment (sebbrudzinski) [inline docs/source/get_started/configure_hipaa.rst]: have to take care

**Comment by sebbrudzinski** [comment]
@pgesek Looks good to me.

**Pull request merged**